### DISCUSSION TITLE
rPi 4G (+GSM) Wi-Fi Hotspot with pi-Hole (and VPN)

### DISCUSSION THREAD

**Discussion by elektrochemikals** [geerlingguy/raspberry-pi-pcie-devices#421] [Ideas]
Hi Jeff,
This idea surely needs improvements, but I had a dream:

Internet (provider) <- 4/5G -> rPi (piHole+VPN)<- (1) eth /or (2)Wi-Fi hotspot-> Wi-Fi router (i.e. MikroTik (VPN?) <- Wi-Fi / eth -> {Home network, NAS, ...}

if the rPi module has GSM -> rPi can periodically inform any desired device of it's dynamic public IP via SMS 
(ie. mobile phone in another country that can use proposed VPN service for country-localized services in the case if rPi is left at home)
or it can be used as 4G modem access point on remote locations - i.e. for soundcloud and to stream rave party from forest :)

also, that rPi device should (can) act like a "GSM tunnel" or "GSM firewall" i.e. sms&call forwarding / blocking / call&sms log / re*khm*odring calls / distort the voice :)

Keep up good work!
All the best!

**Comment by paulwratt** [comment]
there is a dual 1Gb Network with 4G/5G M.2 listed from Waveshare. They also have a multi-IO board with single 1Gb network and PCIe slot. Their wiki shows how to set up "tunnel" and "firewall" for the dual network carrier board (I read it last night). BTW only their recomended 4G module still supports "GSM", with 5G modules that is not a thing any more (here in NZ we dont have GSM anymore).

**Reply by geerlingguy** [reply]
My video next week will cover the dual-Ethernet 5G/4G board, though I haven't gotten the Sierra Wireless 4G Cat12 modem I bought for it working yet. I plan on doing a more in-depth video soon, but here's the one I'm using: https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/367
